Source/HTTP/WinHttp/winhttp_http_task.cpp

```diff
@@ -339,19 +339,38 @@ void winhttp_http_task::_multiple_segment_write_data(_In_ winhttp_http_task* pRe
     const uint64_t defaultChunkSize = 64 * 1024;
     uint64_t safeSize = std::min(pRequestContext->m_requestBodyRemainingToWrite, defaultChunkSize);
 
-    const BYTE* requestBody = nullptr;
-    uint32_t requestBodyBytes = 0;
-    if ((HCHttpCallRequestGetRequestBodyBytes(pRequestContext->m_call, &requestBody, &requestBodyBytes) != S_OK) || 
-        requestBody == nullptr)
+    HCHttpCallRequestBodyReadFunction readFunction = nullptr;
+    uint32_t bodySize = 0;
+
+    if ((HCHttpCallRequestGetRequestBodyReadFunction(pRequestContext->m_call, &readFunction, &bodySize) != S_OK) ||
```

Comment:
is it valid to have no readFunction if bodySize == 0?

Reply:
It won't be used in the case where bodySize == 0 but there should always be a read function

Reply:
I'm not a big fan of using assert, mainly because they disappear in release builds. I'll pass the failed result on to the caller